topics/ui/controls/checkbox.topic

```diff
@@ -0,0 +1,207 @@
+                row:</p>
+            <img src="checkbox_when_to_use_3.png"
+                 alt="A table with checkboxes where the label is placed into the column header" width="706"/>
+                <chapter title="Implementation" collapsible="true" id="checkbox-table-implementation">
+                Checkboxes are rendered in tables with <a
+                    href="%gh-ic%/platform/core-ui/src/ui/BooleanTableCellRenderer.java"><code>BooleanTableCellRenderer</code></a>
+                and edited with <code>DefaultCellEditor(JCheckBox)</code> implementation.
+                For any column that should be rendered as a checkbox, set both a renderer and editor for consistency.
+                The type of data in the correspondent column of the <code>Table</code> model should either be <code>Boolean</code>
+                or <code>String</code> containing <code>true</code> or <code>false</code>.
+            <code-block lang="java">
+                TableColumn column = table.getColumnModel().getColumn(COLUMN_INDEX);
+                column.setCellEditor(JBTable.createBooleanEditor());
+                column.setCellRenderer(new BooleanTableCellRender());
+            </code-block>
+                </chapter>
+        </chapter>
+            <chapter title="Three-state checkbox" id="three-state-checkbox">
+                <p>In a group of options, use the parent checkbox to show the status of its children:</p>
+                <img src="checkbox_three_state.png"
+                     alt="Different states for a parent checkbox: checked, indeterminate, and unchecked" width="706"/>
+                <p>The parent checkbox in checked, indeterminate and unchecked states.</p>
+                    <chapter title="Implementation" collapsible="true">
+                    The three-state checkbox is represented by the <a
+                            href="%gh-ic%/platform/util/ui/src/com/intellij/util/ui/ThreeStateCheckBox.java"><code>ThreeStateCheckBox</code></a>
+                    class which represents its state with the <code>ThreeStateCheckBox.State</code> enum containing <code>SELECTED,
+                        NOT_SELECTED, DONT_CARE</code> states.</chapter>
+                <p>When the user clicks an indeterminate checkbox for the first time, the whole group becomes checked. The
+                    second click unchecks the whole group.</p>
+                <p>An indeterminate checkbox can also show the download status. An example with a remote repository:</p>
+                <img src="checkbox_download_status.png" alt="Indeterminate checkboxes showing download status" width="706"/>
+                <p>
+                    Repositories &quot;tools-base&quot; and &quot;contrib&quot; are being loaded. When loading is
+                    finished, the indeterminate checkbox will be replaced with the checked checkbox if there are
+                    commits, or an unchecked checkbox if there are no commits.
+                </p>
```

Comment:
This was also an image caption in the old article. I'd suggest moving above the image